### PR TITLE
Mc streams

### PR DESCRIPTION
**Pull request by FBRTMaka**
@DanielJMaher

### CODE REVIEW

**Comment by FBRTMaka** [comment]
screen shot 2015-12-06 at 1 56 00 pm

**Comment by DanielJMaher** [comment]
@FBRTMaka This is awesome, but where is the "Old Streams" Button?

File: ooiui/static/js/models/science/StreamModel.js [hunk 1]
```diff
@@ -76,5 +80,53 @@ var StreamCollection = Backbone.Collection.extend({
...
+          });
+          return new StreamCollection(filtered);
+      } else {
+          return this;
+      }
+  },
+  byEndTime: function(binary) {
+
+
+
```
Comment (birdage) [inline ooiui/static/js/models/science/StreamModel.js]: gap

File: ooiui/static/js/models/science/StreamModel.js [hunk 2]
```diff
@@ -76,5 +80,53 @@ var StreamCollection = Backbone.Collection.extend({
...
+      var filtered = this.filter(function (model) {
+          var time = new Date(model.get('end')),
+              timeSinceEnd = new Date().getTime() - time.getTime(),
+              twentyFour = 1000*60*60*24,
```
Comment (birdage) [inline ooiui/static/js/models/science/StreamModel.js]: no calc

File: ooiui/static/js/views/common/PageControlViews.js [hunk 3]
```diff
@@ -51,7 +58,19 @@ template: _.template('<select id="search-param"><option selected>None</selected>
 });
 
 
-var DataCatalogPageControlView = ParentPageControlView.extend({});
+var DataCatalogPageControlView = ParentPageControlView.extend({
+    events: {
+        'change select#search-param': 'click'
+    },
+    template: _.template('<select id="search-param"><option selected>None</selected><% _.each(Object.keys(params), function(key) { %> <optgroup label="<%=key%>"> <% _.each(params[key], function(item){ %> <option id="<%= item[0]  %>"><%= item[1] %></option><% })%></optgroup> <%;}); %></select>'),
```
Comment (birdage) [inline ooiui/static/js/views/common/PageControlViews.js]: partial

File: ooiui/static/js/views/common/StreamQuickView.js [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+            this.$el.html(this.template(this.model.toJSON()));
+        } else {
+            this.$el.html(this.template());
+        }
+        var self = this;
+        setTimeout(function() {
+            self.$el.css({
+                'opacity': 1
+            });
+        }, 0);
```
Comment (birdage) [inline ooiui/static/js/views/common/StreamQuickView.js]: timeout?

File: ooiui/static/js/views/common/StreamQuickView.js [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+        return this;
+    },
+    colorTime: function() {
+        // add a 'freshness' property to the model for easy styling.
+        var time = new Date(this.model.get('end')),
+            timeSinceEnd = new Date().getTime() - time.getTime(),
+            twelve = 1000*60*60*12,
```
Comment (birdage) [inline ooiui/static/js/views/common/StreamQuickView.js]: calc

File: ooiui/static/js/views/common/StreamQuickView.js [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+        $('.modal-backdrop').remove();
+    }
+});
+
+
+var StreamQuickViewItem = ParentView.extend({
+    /*  each stream is handled with this view.
+     */
+    tagName: 'tr',
+    template: _.template('<td><%= display_name %></td>' +
```
Comment (birdage) [inline ooiui/static/js/views/common/StreamQuickView.js]: partial

**Comment by FBRTMaka** [comment]
screen shot 2015-12-07 at 4 26 00 pm

**Pull request merged**